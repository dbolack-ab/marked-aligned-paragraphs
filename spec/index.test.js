import { marked } from 'marked';
import alignedParagraphs from '../src/index.js';

function trimLines(s) {
  return s.split('\n').map(l => l.trim()).join('\n');
}

describe('Aligned Text', () => {
  beforeEach(() => {
    marked.setOptions(marked.getDefaults());
  });

  test('Left Aligned', () => {
    marked.use(alignedParagraphs());
    expect(marked(trimLines(`:- Hello`))).toMatchSnapshot();
  });

  test('Right Aligned', () => {
    marked.use(alignedParagraphs());
    expect(marked(trimLines(`-: Hello`))).toMatchSnapshot();
  });

  test('Center Aligned', () => {
    marked.use(alignedParagraphs());
    expect(marked(trimLines(`:-: Hello`))).toMatchSnapshot();
  });

  test('Ignored inside a code block', () => {
    marked.use(alignedParagraphs());
    expect(marked(trimLines('```\n\n:- Hello\n\n```\n'))).toMatchSnapshot();
  });
});
