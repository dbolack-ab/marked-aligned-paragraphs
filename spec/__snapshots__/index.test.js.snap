// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aligned Text Center Aligned 1`] = `"<p align="Center">Hello</p>"`;

exports[`Aligned Text Ignored inside a code block 1`] = `
"<pre><code>
:- Hello
</code></pre>
"
`;

exports[`Aligned Text Left Aligned 1`] = `"<p align="Left">Hello</p>"`;

exports[`Aligned Text Right Aligned 1`] = `"<p align="Right">Hello</p>"`;
